src/Control/Monad/Ghc.hs

```diff
@@ -60,7 +60,20 @@ instance (MonadIO m, MonadMask m) => MonadMask (GhcT m) where
         wrap g   = GhcT $ GHC.GhcT $ \s -> MTLAdapter (g s)
         unwrap m = unMTLA . GHC.unGhcT (unGhcT m)
 
-    uninterruptibleMask = mask
+    uninterruptibleMask f = wrap $ \s ->
+                              uninterruptibleMask $ \io_restore ->
+                                unwrap (f $ \m -> (wrap $ \s' -> io_restore (unwrap m s'))) s
+      where
+        wrap g   = GhcT $ GHC.GhcT $ \s -> MTLAdapter (g s)
+        unwrap m = unMTLA . GHC.unGhcT (unGhcT m)
+
+    generalBracket acquire release body
+      = wrap $ \s -> generalBracket (unwrap acquire s)
+                                    (\a exitCase -> unwrap (release a exitCase) s)
+                                    (\a -> unwrap (body a) s)
+      where
+        wrap g   = GhcT $ GHC.GhcT $ \s -> MTLAdapter (g s)
+        unwrap m = unMTLA . GHC.unGhcT (unGhcT m)
```

Comment:
This is the main fix: exceptions-0.9 introduced a new method named `generalBracket`, whose type I fixed in exceptions-0.10, so in order to upgrade from exceptions-0.8 to exceptions-0.10, we have to implement this method. If we wanted, we could add an `#ifdef` around this method definition in order to support both.

Reply:
What about depending on exceptions >= 0.8.0?

Reply:
Do you mean >= 0.10.0, or did you change your mind and now want an `#ifdef` in order to support both 0.8 and 0.10?

Reply:
I indeed meant >= 0.10.0 - thanks for spotting that. I would not bother supporting 0.8 and earlier, especially given how it's a v0, and how we're likely one of the last libraries to add support for the new exceptions versions.

Reply:
fixed, merging